sponsorship_compassion/model/contract_group.py

```diff
@@ -23,76 +24,80 @@
                     if 'S' in contract.type:
-                        res[group.id] = True
+                        group.contains_sponsorship = True
                         break
-                else:
-                    res[group.id] = False
             else:
-                res[group.id] = True
-        return res
-
-    _columns = {
-        'contains_sponsorship': fields.function(
-            _contains_sponsorship, string=_('Contains sponsorship'),
-            type='boolean', readonly=True)
-    }
+                group.contains_sponsorship = True
 
-    _defaults = {
-        'contains_sponsorship': lambda self, cr, uid, context: 'S' in
-        context.get('default_type', 'O')
-    }
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
 
-    def generate_invoices(self, cr, uid, ids, invoicer_id=None, context=None):
+    def generate_invoices(self, invoicer=None):
         """ Add birthday gifts generation. """
-        invoicer_id = self._generate_birthday_gifts(cr, uid, ids, invoicer_id,
-                                                    context)
-        invoicer_id = super(contract_group, self).generate_invoices(
-            cr, uid, ids, invoicer_id, context)
-        return invoicer_id
-
-    def _generate_birthday_gifts(self, cr, uid, ids, invoicer_id=None,
-                                 context=None):
+        invoicer = self._generate_birthday_gifts(invoicer)
+        invoicer = super(contract_group, self).generate_invoices(invoicer)
+        return invoicer
+
+    ##########################################################################
+    #                             PRIVATE METHODS                            #
+    ##########################################################################
+
+    @api.multi
+    def _generate_birthday_gifts(self, invoicer=None):
         """ Creates the annual birthday gift for sponsorships that
         have set the option for automatic birthday gift creation. """
         logger.info("Automatic Birthday Gift Generation Started.")
-        if context is None:
-            context = dict()
-        ctx = context.copy()
-        ctx['lang'] = 'en_US'
-        if invoicer_id is None:
-            invoicer_id = self.pool.get('recurring.invoicer').create(
-                cr, uid, {'source': self._name}, ctx)
-        ctx['recurring_invoicer_id'] = invoicer_id
+
+        self.with_context(lang='en_US')
+
+        if invoicer is None:
+            invoicer = self.env['recurring.invoicer'].create(
+                {'source': self._name})
 
         # Search active Sponsorships with automatic birthday gift
-        gen_states = self._get_gen_states()
+        gen_states = self.with_context(
+            recurring_invoicer_id=invoicer)._get_gen_states()

```

Comment:
Is it the id or the record we should put in context ? Please verify and rename context keyword (in recurring_contract as well) or correct the value.